src/intree.js

```diff
@@ -125,7 +129,8 @@ module.exports.setup = function(cfg) {
         let events = task.task.extra.github.events;
         let branches = task.task.extra.github.branches;
         return _.some(events, ev => payload.details['event.type'].startsWith(_.trimEnd(ev, '*'))) &&
-          (!branches || branches && _.includes(branches, payload.details['event.base.repo.branch']));
+          (!branches || branches && _.includes(branches, payload.details['event.base.repo.branch']) ||
+            payload.details['event.type']==='tag');
```

Comment:
Tags are applied to commits , hence tags and branches are not directly related in github. 
`event.base.repo.branch = 'tagname' ` for a tag event. So for a config like below 
```
        events:
          - push
          - tag
        branches:
          - master
```
No task for a tag event will be scheduled. hence I modified the above  check to make sure that branches are ignored for tag events. 
[Sample Payload for a tag push](http://help.appveyor.com/discussions/questions/1050-can-you-skip-a-build-if-there-is-a-tag)

Reply:
I think this is a good fix, but also not setting `event.base.repo.branch` will help.